site/src/pages/OrganizationSettingsPage/OrganizationSettingsPage.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  createOrganization,
+  updateOrganization,
+  deleteOrganization,
+} from "api/queries/organizations";
+import type { UpdateOrganizationRequest } from "api/typesGenerated";
+import { ErrorAlert } from "components/Alert/ErrorAlert";
+import {
+  FormFields,
+  FormSection,
+  HorizontalForm,
+  FormFooter,
+} from "components/Form/Form";
+import { IconField } from "components/IconField/IconField";
+import { Margins } from "components/Margins/Margins";
+import { PageHeader, PageHeaderTitle } from "components/PageHeader/PageHeader";
+import { Stack } from "components/Stack/Stack";
+import {
+  getFormHelpers,
+  nameValidator,
+  displayNameValidator,
+  onChangeTrimmed,
+} from "utils/formUtils";
+import { useOrganizationSettings } from "./OrganizationSettingsLayout";
+
+const MAX_DESCRIPTION_CHAR_LIMIT = 128;
+const MAX_DESCRIPTION_MESSAGE =
+  "Please enter a description that is no longer than 128 characters.";
+
+export const getValidationSchema = (): Yup.AnyObjectSchema =>
+  Yup.object({
+    name: nameValidator("Name"),
+    display_name: displayNameValidator("Display name"),
+    description: Yup.string().max(
+      MAX_DESCRIPTION_CHAR_LIMIT,
+      MAX_DESCRIPTION_MESSAGE,
+    ),
+  });
+
+const OrganizationSettingsPage: FC = () => {
+  const queryClient = useQueryClient();
+  const addOrganizationMutation = useMutation(createOrganization(queryClient));
+  const updateOrganizationMutation = useMutation(
+    updateOrganization(queryClient),
+  );
+  const deleteOrganizationMutation = useMutation(
+    deleteOrganization(queryClient),
+  );
+
+  const { currentOrganizationId, organizations } = useOrganizationSettings();
+
+  const org = organizations.find((org) => org.id === currentOrganizationId)!;
+
+  const error =
+    addOrganizationMutation.error ?? deleteOrganizationMutation.error;
+
+  const form = useFormik<UpdateOrganizationRequest>({
+    initialValues: {
+      name: org.name,
+      display_name: org.display_name,
+      description: org.description,
+      icon: org.icon,
+    },
+    validationSchema: getValidationSchema(),
+    onSubmit: (values) =>
+      updateOrganizationMutation.mutateAsync({ orgId: org.id, req: values }),
+    enableReinitialize: true,
+  });
+  const getFieldHelpers = getFormHelpers(form, error);
+
+  const [newOrgName, setNewOrgName] = useState("");
+
+  return (
+    <Margins verticalMargin={18}>
+      {Boolean(error) && <ErrorAlert error={error} />}
+
+      <PageHeader css={{ paddingTop: 0 }}>
+        <PageHeaderTitle>Organization settings</PageHeaderTitle>
+      </PageHeader>
+
+      <HorizontalForm
+        onSubmit={form.handleSubmit}
+        aria-label="Template settings form"
+      >
+        <FormSection
+          title="General info"
+          description="Change the name or description of the organization."
+        >
+          <FormFields>
+            <TextField
+              {...getFieldHelpers("name")}
+              disabled={form.isSubmitting}
```

Comment:
I see we have this pattern all over the place. We could wrap the fields in a `<fieldset>` and mark it as `disabled` instead to avoid doing it on every field.